mp/src/vgui2/vgui_controls/ProgressBar.cpp

```diff
@@ -76,106 +69,125 @@ void ProgressBar::SetProgressText()
+    int iSubdivCoord = 0 - (m_iSubdivMarksWidth / 2);
+    for (int i = 0; i < m_iSubdivMarksCount; i++)
+    {
+        if (bVertical)
+        {
+            iSubdivCoord += tall / (m_iSubdivMarksCount + 1);
+            surface()->DrawFilledRect(0, iSubdivCoord, wide, iSubdivCoord + m_iSubdivMarksWidth);
+        }
+        else
+        {
+            iSubdivCoord += wide / (m_iSubdivMarksCount + 1);
+            surface()->DrawFilledRect(iSubdivCoord, 0, iSubdivCoord + m_iSubdivMarksWidth, tall);
+        }
+    }
 }
 
-//-----------------------------------------------------------------------------
-// Purpose: 
-//-----------------------------------------------------------------------------
 void ProgressBar::Paint()
 {
-	int wide, tall;
-	GetSize(wide, tall);
-
-	// gaps
-	int segmentTotal = 0, segmentsDrawn = 0;
-	int x = 0, y = 0;
-
-	switch( m_iProgressDirection )
-	{
-	case PROGRESS_WEST:
-		wide -= 2 * m_iBarMargin;
-		x = wide - m_iBarMargin;
-		y = m_iBarInset;
-		segmentTotal = wide / (_segmentGap + _segmentWide);
-		segmentsDrawn = (int)(segmentTotal * _progress);
-		break;
-
-	case PROGRESS_EAST:
+    int wide, tall;
+    GetSize(wide, tall);
+
+    // gaps
+    int segmentTotal = 0, segmentsDrawn = 0;
+    int x = 0, y = 0;
+
+    switch(m_iProgressDirection)
+    {
+    case PROGRESS_WEST:
+        wide -= 2 * m_iBarMargin;
+        x = wide - m_iBarMargin;
+        y = m_iBarInset;
+        segmentTotal = wide / (_segmentGap + _segmentWide);
+        segmentsDrawn = static_cast<int>(segmentTotal * _progress);
+        break;
+
+    case PROGRESS_EAST:
     default:
-		wide -= 2 * m_iBarMargin;
-		x = m_iBarMargin;
-		y = m_iBarInset;
-		segmentTotal = wide / (_segmentGap + _segmentWide);
-		segmentsDrawn = (int)(segmentTotal * _progress);
-		break;
-
-	case PROGRESS_NORTH:
-		tall -= 2 * m_iBarMargin;
-		x = m_iBarInset;
-		y = tall - m_iBarMargin;
-		segmentTotal = tall / (_segmentGap + _segmentWide);
-		segmentsDrawn = (int)(segmentTotal * _progress);
-		break;
-
-	case PROGRESS_SOUTH:
-		tall -= 2 * m_iBarMargin;
-		x = m_iBarInset;
-		y = m_iBarMargin;
-		segmentTotal = tall / (_segmentGap + _segmentWide);
-		segmentsDrawn = (int)(segmentTotal * _progress);
-		break;
-	}
-
-	surface()->DrawSetColor(GetFgColor());
-	for (int i = 0; i < segmentsDrawn; i++)
-	{
-		PaintSegment( x, y, tall, wide );
-	}
+        wide -= 2 * m_iBarMargin;
+        x = m_iBarMargin;
+        y = m_iBarInset;
+        segmentTotal = wide / (_segmentGap + _segmentWide);
+        segmentsDrawn = static_cast<int>(segmentTotal * _progress);
+        break;
+
+    case PROGRESS_NORTH:
+        tall -= 2 * m_iBarMargin;
+        x = m_iBarInset;
+        y = tall - m_iBarMargin;
+        segmentTotal = tall / (_segmentGap + _segmentWide);
+        segmentsDrawn = static_cast<int>(segmentTotal * _progress);
+        break;
+
+    case PROGRESS_SOUTH:
+        tall -= 2 * m_iBarMargin;
+        x = m_iBarInset;
+        y = m_iBarMargin;
```

Comment:
Okay crap I lied. It seems like x and y *were* being set here. But for the other one it actually was 0, 0. Sorry about that! You will want to pass them in again as these values will make the subdiv marks properly paint upon different orientations of the bar.

Reply:
They were being set but I passed in `0, 0` at the time anyways. Didn't draw correctly with `x` and `y` passed in as they are modified in `PaintSegment`. I suppose though this wouldn't work with different insets.

Reply:
Ohh right! Derp.